analyzers/src/SonarAnalyzer.CSharp/Rules/InvocationResolvesToOverrideWithParams.cs

```diff
@@ -18,144 +18,106 @@
-                .Select(typeInfo => typeInfo.Type ?? typeInfo.ConvertedType) // Action and Func won't always resolve properly with Type
-                .ToList();
-            if (argumentTypes.Any(type => type is IErrorTypeSymbol))
+    protected override void Initialize(SonarAnalysisContext context)
+    {
+        context.RegisterNodeAction(
+            c =>
             {
-                return;
-            }
-
-            var possibleOtherMethods = invokedMethodSymbol.ContainingType.GetMembers(invokedMethodSymbol.Name)
-                .OfType<IMethodSymbol>()
-                .Where(m => !m.IsVararg)
-                .Where(m => m.MethodKind == invokedMethodSymbol.MethodKind)
-                .Where(m => !invokedMethodSymbol.Equals(m))
-                .Where(m => m.Parameters.Any() && !m.Parameters.Last().IsParams);
-
-            var otherMethod = possibleOtherMethods.FirstOrDefault(possibleOtherMethod =>
-                    ArgumentsMatchParameters(
-                        argumentList,
-                        argumentTypes.Select(t => t as INamedTypeSymbol).ToList(),
-                        possibleOtherMethod,
-                        context.SemanticModel));
+                var invocation = (InvocationExpressionSyntax)c.Node;
+                CheckCall(c, invocation, invocation.ArgumentList);
+            },
+            SyntaxKind.InvocationExpression);
 
-            if (otherMethod != null)
+        context.RegisterNodeAction(
+            c =>
             {
-                context.ReportIssue(Diagnostic.Create(
-                    Rule,
-                    node.GetLocation(),
-                    otherMethod.ToMinimalDisplayString(context.SemanticModel, node.SpanStart)));
-            }
-        }
+                var objectCreation = (ObjectCreationExpressionSyntax)c.Node;
+                CheckCall(c, objectCreation, objectCreation.ArgumentList);
+            },
+            SyntaxKind.ObjectCreationExpression);
+    }
 
-        private static bool IsInvocationWithExplicitArray(ArgumentListSyntax argumentList, IMethodSymbol invokedMethodSymbol, SemanticModel semanticModel)
+    private static void CheckCall(SonarSyntaxNodeReportingContext context, SyntaxNode node, ArgumentListSyntax argumentList)
+    {
+        if (argumentList is { Arguments.Count: > 0 }
+            && context.SemanticModel.GetSymbolInfo(node).Symbol is IMethodSymbol method
+            && method.Parameters.LastOrDefault() is { IsParams: true }
+            && !IsInvocationWithExplicitArray(argumentList, method, context.SemanticModel)
+            && ArgumentTypes(context, argumentList) is var argumentTypes
+            && argumentTypes.All(x => x is not IErrorTypeSymbol)
+            && OtherOverloadsOf(method).FirstOrDefault(x =>
+                ArgumentsMatchParameters(argumentList, argumentTypes, x, context.SemanticModel) && MethodAccessibleWithinType(x, context.ContainingSymbol.ContainingType))
+                    is { } otherMethod)
```

Comment:
This is too much of nesting - there's too much lines that do not start with `&&` as a sign that we should do it differently. Let's extract the condition into a local method.
```suggestion
            && OtherOverloadsOf(method).FirstOrDefault(IsPossibleMatch) is { } otherMethod)
```